src/masternodeman.cpp

```diff
@@ -776,3 +836,109 @@ void CMasternodeMan::UpdateLastPaid(const CBlockIndex *pindex) {
     // every time is like the first time if winners list is not synced
     IsFirstRun = !masternodeSync.IsWinnersListSynced();
 }
+
+void CMasternodeMan::UpdateWatchdogVoteTime(const CTxIn& vin)
+{
+    LOCK(cs);
+    CMasternode* pMN = Find(vin);
+    if(!pMN)  {
+        return;
+    }
+    pMN->UpdateWatchdogVoteTime();
+    nLastWatchdogVoteTime = GetTime();
+}
+
+bool CMasternodeMan::IsWatchdogActive()
+{
+    LOCK(cs);
+    // Check if any masternodes have voted recently, otherwise return false
+    if((GetTime() - nLastWatchdogVoteTime) > MASTERNODE_WATCHDOG_MAX_SECONDS) {
+        return false;
+    }
+    return true;
+}
+
+void CMasternodeMan::AddGovernanceVote(const CTxIn& vin, uint256 nGovernanceObjectHash)
+{
+    LOCK(cs);
+    CMasternode* pMN = Find(vin);
+    if(!pMN)  {
+        return;
+    }
+    pMN->AddGovernanceVote(nGovernanceObjectHash);
+}
+
+void CMasternodeMan::RemoveGovernanceObject(uint256 nGovernanceObjectHash)
+{
+    LOCK(cs);
+    BOOST_FOREACH(CMasternode& mn, vMasternodes) {
+        mn.RemoveGovernanceObject(nGovernanceObjectHash);
+    }
+}
+
+void CMasternodeMan::CheckMasternode(const CTxIn& vin, bool fForce)
+{
+    LOCK(cs);
+    CMasternode* pMN = Find(vin);
+    if(!pMN)  {
+        return;
+    }
+    pMN->Check(fForce);
+}
+
+void CMasternodeMan::CheckMasternode(const CPubKey& pubKeyMasternode, bool fForce)
+{
+    LOCK(cs);
+    CMasternode* pMN = Find(pubKeyMasternode);
+    if(!pMN)  {
+        return;
+    }
+    pMN->Check(fForce);
+}
+
+int CMasternodeMan::GetMasternodeState(const CTxIn& vin)
+{
+    LOCK(cs);
+    CMasternode* pMN = Find(vin);
+    if(!pMN)  {
+        return CMasternode::MASTERNODE_REMOVE;
+    }
+    return pMN->nActiveState;
+}
+
+int CMasternodeMan::GetMasternodeState(const CPubKey& pubKeyMasternode)
+{
+    LOCK(cs);
+    CMasternode* pMN = Find(pubKeyMasternode);
+    if(!pMN)  {
+        return CMasternode::MASTERNODE_REMOVE;
+    }
+    return pMN->nActiveState;
+}
+
+bool CMasternodeMan::IsMasternodePingedWithin(const CTxIn& vin, int nSeconds, int64_t nTimeToCheckAt)
+{
+    LOCK(cs);
+    CMasternode* pMN = Find(vin);
+    if(!pMN)  {
```

Comment:
I fixed the other items.  For this one I believe that my version is clearer and more straightforward.